spec/models/machine_spec.rb

```diff
@@ -54,7 +54,7 @@
   describe '#components_attributes=' do
     it 'should not raise when component is valid' do
       machine = build(:machine, account:, components_attributes: [
-        attributes_for(:component, account:),
+        attributes_for(:component, account:, environment: nil),
```

Comment:
Why doesn't `attributes_for` run through the factory's `initialize_with`? According to the docs, [it should](https://github.com/thoughtbot/factory_bot/blob/bd0a10c5ab9f93425beeba2ab17e503dceb368ab/docs/src/ref/build-strategies.md?plain=1#L55C1-L59).

Reply:
The problem here is that the `NIL_ACCOUNT` and `NIL_ENVIRONMENT` sentinel values aren't filtered out of the resulting attribute hash, which ofc causes issues during record initialization. Maybe we can patch `attributes_for` to reject those values like we are in `initialize_with`?

Reply:
Meh. Not worth the effort.

Reply:
Not worth the effort of a workaround, but I opened an issue: https://github.com/thoughtbot/factory_bot/issues/1606